vendor/k8s.io/kubernetes/openshift-kube-apiserver/admission/customresourcevalidation/proxy/validate_proxy.go

```diff
@@ -0,0 +1,286 @@
+	}
+
+	allErrs = append(allErrs, validateProxySpec(obj.Spec)...)
+
+	return allErrs
+}
+
+func (proxyV1) ValidateUpdate(uncastObj runtime.Object, uncastOldObj runtime.Object) field.ErrorList {
+	obj, allErrs := toProxyV1(uncastObj)
+	if len(allErrs) > 0 {
+		return allErrs
+	}
+	oldObj, allErrs := toProxyV1(uncastOldObj)
+	if len(allErrs) > 0 {
+		return allErrs
+	}
+
+	allErrs = append(allErrs, validation.ValidateObjectMetaUpdate(&obj.ObjectMeta, &oldObj.ObjectMeta, field.NewPath("metadata"))...)
+	allErrs = append(allErrs, validateProxySpec(obj.Spec)...)
+
+	return allErrs
+}
+
+// validateReadinessEndpoint validates a proxy readinessendpoint.
+func validateReadinessEndpoint(endpoint string) error {
+	err := validateReadinessEndpointWithRetries(endpoint, maxRetries)
+		if err != nil {
+			return err
+		}
+
+	return nil
+}
+
+// validateReadinessEndpointWithRetries tries to validate the proxy readinessendpoint in a finite loop,
+// it returns the last result if it never succeeds.
+func validateReadinessEndpointWithRetries(endpoint string, retries int) field.ErrorList {
+	var err error
+	for i := 0; i < retries; i++ {
+		result, output, err = runHTTPReadinessProbe(endpoint)
+		if err == nil {
+			return nil
+		}
+	}
+	return err
+}
+
+// runHTTPReadinessProbe issues an http GET request to endpoint and returns an error
+// if a 2XX or 3XX http status code is not returned.
+func runHTTPReadinessProbe(endpoint string) error {
+	url, err := url.Parse(endpoint)
+	if err != nil {
+		return err
+	}
+	timeout := time.Duration(5) * time.Second
+	resp, err := http.Get(url.String())
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode >= http.StatusOK && resp.StatusCode < http.StatusBadRequest {
+		return nil
+	}
+
+	return fmt.Sprintf("HTTP probe failed with statuscode: %d", resp.StatusCode)
+}
+
+// createNoProxy combines user-provided & platform-specific values to create a comma-separated
+// list of unique NO_PROXY values. Platform values are: serviceCIDR, podCIDR, localhost,
+// 127.0.0.1, api.clusterdomain, api-int.clusterdomain, etcd-idx.clusterdomain
+// If platform is not vSphere or None add 169.254.169.254 to the list of NO_PROXY
+// address. We should not proxy the instance metadata services:
+// https://docs.openstack.org/nova/latest/user/metadata.html
+// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html
+// https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service
+// https://cloud.google.com/compute/docs/storing-retrieving-metadata
+func createNoProxy(spec *configv1.ProxySpec) (string, error) {
+	// TODO: Create client with configv1 scheme registered.
+	// Retrieve the cluster infrastructure config.
+	infraConfig := &configv1.Infrastructure{}
+	err = kubeClient.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, infraConfig)
+	if err != nil {
+		return "", err
+	}
+	apiServerURL, err := url.Parse(getAPIServerURL(infraConfig.Status.APIServerURL))
```

Comment:
this is the LB domain, right? Are we sure we don't need a proxy for that? I expect this to depend a lot on the networking environment /cc @wking @squeed